src/configure/mod.rs

```diff
@@ -525,17 +534,26 @@ fn get_offline_autoconfig(context: &Context, param: &LoginParam) -> Option<Login
 
 fn try_imap_connections(
     context: &Context,
-    mut param: &mut LoginParam,
+    param: &mut LoginParam,
+    original_param: &LoginParam,
+    // original_param is used to check whether a particular setting was set manually by the user.
+    // If yes, we do not want to change it to avoid confusing error messages
+    // (you set port 443, but the app tells you it couldn't connect on port 993).
     was_autoconfig: bool,
 ) -> bool {
     // progress 650 and 660
-    if let Some(res) = try_imap_connection(context, &mut param, was_autoconfig, 0) {
+    if let Some(res) = try_imap_connection(context, param, original_param, was_autoconfig, 0) {
         return res;
     }
     progress!(context, 670);
-    param.server_flags &= !(DC_LP_IMAP_SOCKET_FLAGS);
-    param.server_flags |= DC_LP_IMAP_SOCKET_SSL;
-    param.mail_port = 993;
+
+    if original_param.server_flags & DC_LP_IMAP_SOCKET_FLAGS == 0 {
+        param.server_flags &= !(DC_LP_IMAP_SOCKET_FLAGS);
+        param.server_flags |= DC_LP_IMAP_SOCKET_SSL;
+    }
+    if original_param.mail_port == 0 {
+        param.mail_port = 993;
+    }
```

Comment:
If one of DC_LP_IMAP_SOCKET* or mail_port is set, we still vary the other, which makes sense beecause it can even make configurations succeed that failed before.